backend/vecsim_app/main.py

```diff
@@ -55,21 +45,37 @@ async def startup():
 app.mount(
     path="/", app=SinglePageApplication(directory=gui_build_dir), name="SPA"
 )
-
 if __name__ == "__main__":
+    import logging
     import os
+
+    logging.basicConfig(level=logging.INFO)
+
     env = os.environ.get("DEPLOYMENT", "prod")
+    logging.info(f"Running in {env} mode")
 
     server_attr = {
-        "host": "0.0.0.0",
+        "host": config.SERVER_HOST,
         "reload": True,
-        "port": 8888,
-        "workers": 1
+        "port": int(config.SERVER_PORT),
+        "workers": 1,
+        "log_level": "debug",
     }
     if env == "prod":
-        server_attr.update({"reload": False,
-                            "workers": 2,
-                            "ssl_keyfile": "key.pem",
-                            "ssl_certfile": "full.pem"})
+        server_attr.update(
+            {
+                "reload": False,
+                "workers": 2,
+                "ssl_keyfile": "key.pem",
+                "ssl_certfile": "full.pem",
+            }
+        )
+        app.add_middleware(
+            CORSMiddleware,
+            allow_origins="*",
+            allow_credentials=True,
+            allow_methods=["*"],
+            allow_headers=["*"]
+        )
 
-    uvicorn.run("main:app", **server_attr)
+    uvicorn.run("vecsim_app.main:app", **server_attr)
```

Comment:
this should be able to run from `./backend` dir where we've installed pip package `vecsim_app`